system/services/validation/EmailCenterValidators.cfc

```diff
@@ -41,23 +48,38 @@ component {
 	public array function existingEmailsUsingInvalidDomains( required string validDomains ) {
 		var delims    = ", #Chr( 10 )##Chr( 13 )#";
 		var domains   = ListToArray( Trim( arguments.validDomains ), delims );
-		var addresses = $getPresideObject( "email_template" ).selectData( distinct=true, selectFields=[ "from_address" ] );
+		var addresses = $getPresideObject( "email_template" ).selectData(
+			  distinct     = true
+			, selectFields = [ "id","from_address" ]
+		);
 		var badaddresses = [];
 
 		for( var address in addresses ) {
 			if ( Len( address.from_address) ) {
-				var domain = _getDomainFromEmail( address.from_address );
-				if ( !ArrayFindNoCase( domains, domain ) ) {
-					ArrayAppend( badaddresses, address.from_address );
+				if ( _getSystemEmailTemplateService().templateExists( address.id ) ) {
```

Comment:
We should only run this check where the email is marked as a system email. Otherwise, we'll be unsetting the from address for all editorial emails with a from address I think?!